File: app/src/main/java/com/example/myapplication/core/profile/ProfileGoalValidator.kt
package com.example.myapplication.core.profile

import com.example.myapplication.core.model.FitnessGoal

object ProfileGoalValidator {
    fun validate(profile: PersonalProfile, fitnessGoal: FitnessGoal): List<String> {
        if (!profile.currentWeightKg.isFinite() || !profile.targetWeightKg.isFinite()) {
            return emptyList()
        }

        return when (fitnessGoal) {
            FitnessGoal.MUSCLE_GAIN -> if (profile.targetWeightKg <= profile.currentWeightKg) {
                listOf("Mục tiêu tăng cơ cần cân nặng mục tiêu cao hơn cân nặng hiện tại.")
            } else {
                emptyList()
            }

            FitnessGoal.FAT_LOSS_CONDITIONING -> if (profile.targetWeightKg >= profile.currentWeightKg) {
                listOf("Mục tiêu giảm mỡ cần cân nặng mục tiêu thấp hơn cân nặng hiện tại.")
            } else {
                emptyList()
            }

            FitnessGoal.ENDURANCE,
            FitnessGoal.GENERAL_FITNESS,
            -> emptyList()
        }
    }
}

File: flutter/lib/core/profile/profile_goal_validator.dart
import '../model/goal_models.dart';
import '../model/profile_models.dart';

class ProfileGoalValidator {
  static List<String> validate({
    required PersonalProfile profile,
    required FitnessGoal fitnessGoal,
  }) {
    if (!profile.currentWeightKg.isFinite || !profile.targetWeightKg.isFinite) {
      return const [];
    }

    switch (fitnessGoal) {
      case FitnessGoal.muscleGain:
        if (profile.targetWeightKg <= profile.currentWeightKg) {
          return const ["Mục tiêu tăng cơ cần cân nặng mục tiêu cao hơn cân nặng hiện tại."];
        }
        return const [];
      case FitnessGoal.fatLossConditioning:
        if (profile.targetWeightKg >= profile.currentWeightKg) {
          return const ["Mục tiêu giảm mỡ cần cân nặng mục tiêu thấp hơn cân nặng hiện tại."];
        }
        return const [];
      case FitnessGoal.endurance:
      case FitnessGoal.generalFitness:
        return const [];
    }
  }
}
